src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green ' | 'blue';
 
+export interface Figure {
+  shape: Shape
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  color: Color;
+
+  constructor(
+    color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || b + c <= a || a + c <= b) {
+      throw new Error('Invalid triangle sides');
+    }
 
+    this.color = color;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = (Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)))
```

Comment:
- Avoid one-letter names for variables.
- you get `this.a` and others twice, you can use destructing to avoid repeating `const { a, b, c } = this;